### ISSUE TITLE
bug: attention_proportional sums normalized probabilities, so prefill allocation cannot distinguish layers

### ISSUE THREAD

**Issue by umran666** [umran666/UniqKache#65]
### Bug and location

[src/uniqkache/allocation/attention_proportional.py:48-62](https://github.com/umran666/UniqKache/blob/0b878a1acc3f2a84ea0ff16194456b8ed10191e8/src/uniqkache/allocation/attention_proportional.py#L48-L62) weights layers by `cum_attention.sum()`, intending to capture differences in attention concentration. But `reduce_attention()` averages normalized attention probabilities across queries/heads/batches. Summing the resulting key vector is one for every layer after one prefill, regardless of concentration.

The runtime records that prefill attention and immediately allocates at [src/uniqkache/runtime/generation.py:203-213](https://github.com/umran666/UniqKache/blob/0b878a1acc3f2a84ea0ff16194456b8ed10191e8/src/uniqkache/runtime/generation.py#L203-L213).

### Reproduction

```python
import torch
from uniqkache import CacheConfig, KVCache
from uniqkache.allocation import AttentionProportionalAllocationStrategy

cache = KVCache(CacheConfig(2, 1, 4, dtype=torch.float32))
kv = torch.ones(1, 1, 4, 4)
for layer in range(2):
    cache.append(layer, kv, kv)
cache.note_attention(0, torch.tensor([[[[1.0, 0.0, 0.0, 0.0]]]]))
cache.note_attention(1, torch.tensor([[[[0.25, 0.25, 0.25, 0.25]]]]))
print([float(x.metadata.cum_attention.sum()) for x in cache.store.layers])
print(AttentionProportionalAllocationStrategy().allocate(16, cache))
```

Observed: attention totals `[1.0, 1.0]` and allocation `[8, 8]`, despite maximally concentrated versus uniform attention. The same normalization identity applies to other distributions; small allocation differences can come from rounding instead.

### Impact and expected behavior

The advertised attention-responsive layer allocation is effectively uniform at the runtime decision point and cannot measure its stated signal. Tests using manually assigned unequal cumulative totals miss this issue.

Use a layer statistic that actually distinguishes the intended attention property, or correct the exposed strategy and its claims. Verify the behavior with normalized model attention at the real prefill allocation boundary. This remains relevant independently of the list-capacity crash in #48.

### Verification

Reviewed commit: `0b878a1acc3f2a84ea0ff16194456b8ed10191e8`. Reproduced on Windows with Python 3.10 and PyTorch 2.6.0+cu124; CPU unless otherwise specified. Synthetic-model results below are correctness diagnostics, not language-model quality claims.
